pkg/collectsub/client/mock.go

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
```

Comment:
Why not use mockgen or https://github.com/traefik/mocktail?

Reply:
ooo this is very very cool! definitely something i should check out, in this case, i didn't need to fully mock it but more just have a way to exercise this for another PR that I'm working on for #244 . Perhaps let's get this in first to unblock and then let's have a follow-up PR for the mocking! It looks like we can do something similar with https://pkg.go.dev/github.com/stretchr/testify/mock that it uses without the external dependency on tooling as well (which i'm less keen of using tooling that isn't at least 1.0 and mature).

Reply:
We have been using mockgen in Scorecard and other Projects. I agree, this is not a blocker. 👍 Thanks